core/domain/auth_domain.py

```diff
@@ -126,9 +135,9 @@ class UserAuthDetails:
     def __init__(
         self,
         user_id: str,
-        gae_id: str,
-        firebase_auth_id: str,
-        parent_user_id: str,
+        gae_id: Optional[str],
+        firebase_auth_id: Optional[str],
+        parent_user_id: Optional[str],
```

Comment:
I think this should be Optional[str] not str, because for profile user `gae_id` and `firebase_auth_id` is None, and for full user `parent_user_id` is None. Also, this same thing is mentioned in `UserAuthDetailsModel`.
https://github.com/oppia/oppia/blob/bd159ef1fa54d875e6c78551dcbbf5c518565508/core/storage/auth/gae_models.py#L45

Also, looking at the validate() function it supports the fact that these arguments should be Optional[str].
https://github.com/oppia/oppia/blob/bd159ef1fa54d875e6c78551dcbbf5c518565508/core/domain/auth_domain.py#L194